apps/arm/ArmConfig.h

```diff
@@ -12,155 +12,71 @@ namespace ArmConfig {
     /*** ENCODER CONFIG ***/
 
     static Encoder::t_encoderConfig turnTableEncoderConfig = {
-        .pin_PWM = ENC_PWM_TRNTBL,
-        .degreesPerUnit = 360,
-        .zeroOffset_Degrees = 0.0
+        ENC_PWM_TRNTBL, NC, NC, NC, NC, NC, NC, 360.0, 0, Encoder::x2_encoding
```

Comment:
I may have been using incorrect syntax before, but is there a correct syntax with named parameters? It makes it much more clear what each member is.

Reply:
you weren't using incorrect syntax, just syntax only available in C99 or in C++20(released like 2 weeks ago officially or smth lol), and we use c++14 so wPedantic will yell at u for using it.
See this: https://stackoverflow.com/questions/6181715/convenient-c-struct-initialisation

not sure what to do here because its in a header file so i can't just run it thru the c compiler instead, like i could if it was in a c file.

Reply:
@wmmc88 not sure if i want to merge this in like this, it makes the more complicated configuration structs pretty hard to decipher. Will it compile with `-std=c++20`

Reply:
It compiles regardless of the std flag. but its is not "following the standard". `wPedantic` is what causes the pipeline failure. See this: https://stackoverflow.com/questions/37587318/why-compiling-with-c11-is-successful-despite-using-c14-feature

the `-std=c++20` flag is actually not implemented yet. Maybe in the next release. it has `std=c++2a` but my understanding is that its like a "use at ur own risk. this is unstable beta" type option.

Reply:
@jetkov @qinyang-bao i think from our discussion the other day, we should just use `std=c++2a` and revert to the old initialization method.

Reply:
Has this change been made to the Makefile? I don't think we should make this change in this PR...

Reply:
Yeah if you could test it in this repo but implement it in a separate PR that would be great

Reply:
Tested, if this flag is used instead of gnu++14, compilation would fail

Reply:
Using gnu++2a instead of c++2a works though